GrandCentralBoard/Widgets/Harvest/HarvestWidgetBuilder.swift

```diff
@@ -14,8 +14,16 @@ final class HarvestWidgetBuilder: WidgetBuilding {
 
     func build(json: AnyObject) throws -> WidgetControlling {
         let settings = try HarvestWidgetSettings.decode(json)
-        let harvestAPI = HarvestAPI(account: settings.account, refreshCredentials: settings.refreshCredentials,
-                                     downloader: settings.downloader, numberOfDaysToFetch: settings.numberOfDays)
+        let today = NSDate()
+        let billableDates = BillableDates(referenceDate: today,
+                                          numberOfPreviousDays: settings.numberOfDays,
+                                          includeWeekends: settings.includeWeekends ?? false)
+
+        let harvestAPI = HarvestAPI(account: settings.account,
```

Comment:
You are providing billable dates only once, during widget creation. So if the widget works for two days, you will still fetch the same (old) days.